src/main/java/com/microsoft/lst_bench/task/custom/SkipFailedQueryTaskExecutor.java

```diff
@@ -0,0 +1,111 @@
+import com.microsoft.lst_bench.exec.FileExec;
+import com.microsoft.lst_bench.exec.StatementExec;
+import com.microsoft.lst_bench.exec.TaskExec;
+import com.microsoft.lst_bench.telemetry.EventInfo.Status;
+import com.microsoft.lst_bench.telemetry.SQLTelemetryRegistry;
+import com.microsoft.lst_bench.util.StringUtils;
+import java.time.Instant;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Custom task executor implementation that allows users to avoid failing execution by skipping
+ * queries which return a specific substring in their error message. For this task executor, we
+ * allow users to determine the exception strings as part of the input, by specifying parameter
+ * 'skip_failed_query_task_strings'. If multiple strings can lead to a skip action, they need to be
+ * separated with delimiter ';' by default.
+ */
+public class SkipFailedQueryTaskExecutor extends CustomTaskExecutor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DependentTaskExecutor.class);
+
+  private final String SKIP_FAILED_QUERY_TASK_DELIMITER = ";";
+  private final String SKIP_FAILED_QUERY_TASK_STRINGS = "skip_failed_query_task_strings";
+
+  public SkipFailedQueryTaskExecutor(
+      SQLTelemetryRegistry telemetryRegistry,
+      String experimentStartTime,
+      Map<String, String> arguments) {
+    super(telemetryRegistry, experimentStartTime, arguments);
+  }
+
+  @Override
+  public void executeTask(Connection connection, TaskExec task, Map<String, Object> values)
+      throws ClientException {
+    String[] exceptionTaskStrings;
+    if (this.getArguments() == null
+        || this.getArguments().get(SKIP_FAILED_QUERY_TASK_STRINGS) == null) {
+      exceptionTaskStrings = new String[] {""};
+    } else {
+      exceptionTaskStrings =
+          this.getArguments()
+              .get(SKIP_FAILED_QUERY_TASK_STRINGS)
+              .split(SKIP_FAILED_QUERY_TASK_DELIMITER);
+    }
+
+    for (FileExec file : task.getFiles()) {
+      Instant fileStartTime = Instant.now();
+      try {
+        for (StatementExec statement : file.getStatements()) {
+          boolean skip = false;
+          Instant statementStartTime = Instant.now();
+          try {
+            connection.execute(StringUtils.replaceParameters(statement, values).getStatement());
+          } catch (Exception e) {
+            String loggedError =
+                "Exception executing statement: "
+                    + statement.getId()
+                    + ", statement text: "
+                    + statement.getStatement()
+                    + "; error message: "
+                    + e.getMessage();
+            for (String skipException : exceptionTaskStrings) {
+              if (e.getMessage().contains(skipException)) {
+                LOGGER.warn(loggedError);
+                skip = true;
+                break;
+              }
+            }
+            writeStatementEvent(
```

Comment:
@jcamachor: Do you think it might make sense to extend the status enum to include something that indicates a failure but that we continued with the execution instead of terminating it?

Reply:
I have the feeling this is something very specific from this TaskExecutor, if there are no core scenarios when this could happens I would not change it.

Reply:
Maybe having a `WARN` status would be useful? That way is a little bit more generic. But I agree making the differentiation in the telemetry that we capture is useful (and maybe even in the logging message).